# Latihan List

submitted by Marsetio Noorprajuda

## Membuat list

* 1. Buatlah list dengan value 10 sampai 15. Simpan ke dalam variable dengan nama *my_list*
* 2. Buatlah list dengan range 100 hingga 150 dengan interval 5. SImpan ke dalam variable dengan nama *my_list_2*

In [5]:
my_list = [10, 11, 12, 13, 14, 15]
my_list

[10, 11, 12, 13, 14, 15]

In [7]:
my_list_2 = list(range(100,151,5))
my_list_2

[100, 105, 110, 115, 120, 125, 130, 135, 140, 145, 150]

*hint*

In [31]:
# Jawaban no. 1

my_list = [10, 11, 12, 13, 14, 15]

#atau

my_list = list(range(10, 16 ))

my_list

[10, 11, 12, 13, 14, 15]

In [32]:
# Jawaban no. 2

my_list_2 = list(range(100, 151, 5))
my_list_2

[100, 105, 110, 115, 120, 125, 130, 135, 140, 145, 150]

## Subsetting & Slicing list

* 1. dari my_list, pilih angka 13 dengan metode subsetting
* 2. dari my_list, pilih angka 14 dengan metode subsetting
* 3. dari my_list_2, pilih angka dari 120 hingga 145 dengan metode slicing
* 4. dari my_list_2, pilih angka dari 150 hingga 125 dengan metode slicing

In [33]:
print(my_list[3])

13


In [10]:
print(my_list[4])

14


In [13]:
print(my_list_2[4:-1])

[120, 125, 130, 135, 140, 145]


In [34]:
print(sorted(my_list_2[5:11],reverse=True))

[150, 145, 140, 135, 130, 125]


*hint*

In [6]:
# Jawaban no. 1

my_list[3]

13

In [8]:
# Jawaban no. 2
my_list[-2]

# atau

my_list[4]

14

In [10]:
# Jawaban no. 3
my_list_2[4:10]

[120, 125, 130, 135, 140, 145]

In [15]:
# Jawaban no. 4
my_list_2[5:]

[125, 130, 135, 140, 145, 150]

## Manipulasi list

* 1. Gabungkan (concatenate) my_list dan my_list_2, dan simpan menjadi my_list_3
* 2. Ganti angka 100 pada my_list_3 dengan 1000
* 3. Tambahkan nilai 155 sebagai element terakhir pada my_list_3 
* 4. Tambahkan nilai 1500 pada index ke-4 di my_list_3
* 5. sorting nilai yang ada pada my_list_3, dari terbesar ke terkecil (descending)
* 6. Hapus element terakhir yang ada pada my_list_3
* 7. Hapus nilai 1000 dan 1500 dari my_list_3
* 8. Cek ada berapa panjang element pada my_list_3?

In [88]:
my_list_3 = my_list + my_list_2
my_list_3 

[10, 11, 12, 13, 14, 15, 100, 105, 110, 115, 120, 125, 130, 135, 140, 145, 150]

In [89]:
for i in range(len(my_list_3)):
    if my_list_3[i] == 100:
        my_list_3[i] = 1000
        
print(my_list_3)

[10, 11, 12, 13, 14, 15, 1000, 105, 110, 115, 120, 125, 130, 135, 140, 145, 150]


In [90]:
my_list_3.append(155)
my_list_3

[10,
 11,
 12,
 13,
 14,
 15,
 1000,
 105,
 110,
 115,
 120,
 125,
 130,
 135,
 140,
 145,
 150,
 155]

In [91]:
my_list_3.insert(3, 1500)
my_list_3

[10,
 11,
 12,
 1500,
 13,
 14,
 15,
 1000,
 105,
 110,
 115,
 120,
 125,
 130,
 135,
 140,
 145,
 150,
 155]

In [92]:
my_list_3.sort(reverse=True)
my_list_3

[1500,
 1000,
 155,
 150,
 145,
 140,
 135,
 130,
 125,
 120,
 115,
 110,
 105,
 15,
 14,
 13,
 12,
 11,
 10]

In [93]:
my_list_3.pop()
my_list_3

[1500,
 1000,
 155,
 150,
 145,
 140,
 135,
 130,
 125,
 120,
 115,
 110,
 105,
 15,
 14,
 13,
 12,
 11]

In [94]:
my_list_3.remove(1500)
my_list_3.remove(1000)

    
my_list_3
        

[155, 150, 145, 140, 135, 130, 125, 120, 115, 110, 105, 15, 14, 13, 12, 11]

In [95]:
print(len(my_list_3))

16


*hint*

In [22]:
# Jawaban no. 1

my_list_3 = my_list + my_list_2
my_list_3

[10, 11, 12, 13, 14, 15, 100, 105, 110, 115, 120, 125, 130, 135, 140, 145, 150]

In [23]:
# Jawaban no. 2
my_list_3[6] = 1000
my_list_3

[10,
 11,
 12,
 13,
 14,
 15,
 1000,
 105,
 110,
 115,
 120,
 125,
 130,
 135,
 140,
 145,
 150]

In [24]:
# Jawaban no. 3
my_list_3.append(155)
my_list_3

In [25]:
# Jawaban no. 4
my_list_3.insert(4, 1500)
my_list_3

[10,
 11,
 12,
 13,
 1500,
 14,
 15,
 1000,
 105,
 110,
 115,
 120,
 125,
 130,
 135,
 140,
 145,
 150,
 155]

In [28]:
# Jawaban no. 5
my_list_3.sort(reverse = True)
my_list_3

[1500,
 1000,
 155,
 150,
 145,
 140,
 135,
 130,
 125,
 120,
 115,
 110,
 105,
 15,
 14,
 13,
 12,
 11,
 10]

In [29]:
# Jawaban no. 6
my_list_3.pop()
my_list_3

[1500,
 1000,
 155,
 150,
 145,
 140,
 135,
 130,
 125,
 120,
 115,
 110,
 105,
 15,
 14,
 13,
 12,
 11]

In [30]:
# Jawaban no. 7
my_list_3.remove(1000)
my_list_3.remove(1500)
my_list_3

[155, 150, 145, 140, 135, 130, 125, 120, 115, 110, 105, 15, 14, 13, 12, 11]

In [31]:
len(my_list_3)

16

## List Operations
Dari my_list_3, cari beberapa nilai di bawah ini:

- Nilai minimum
- Nilai maximum

In [96]:
min(my_list_3)

11

In [97]:
max(my_list_3)

155

*hint*

In [33]:
min(my_list_3)

11

In [34]:
max(my_list_3)

155

# Latihan Numpy

## Membuat Numpy

1. import numpy library as np
2. Buatlah 1D array dengan value: 100, 200, 300, 400, 500. Simpan ke dalam variable dengan nama *my_array*
3. Buatlah 1D array dengan range value dari 20 hingga 80, dengan interval 5. Simpan ke dalam variable dengan nama *my_array_2*
4. Buatlah 1D array dengan value zero (0), dengan panjang element 5. Simpan ke dalam variable dengan nama *array_zero*
4. Buatlah 1D array dengan value  (1), dengan panjang element 5. Simpan ke dalam variable dengan nama *array_one*
4. Buatlah 1D array dengan bilangan desimal random antara 0 dan 1, dengan panjang element 8. Simpan ke dalam variable dengan nama *array_random*
4. Buatlah 1D array dengan bilangan bulat random antara 10 dan 100, dengan panjang element 8. Simpan ke dalam variable dengan nama *array_integer*

In [98]:
import numpy as np

In [99]:
my_array = np.array(list(range(100,501,100)))
my_array

array([100, 200, 300, 400, 500])

In [102]:
my_array_2 = np.array(list(range(20,80,5)))
my_array_2

array([20, 25, 30, 35, 40, 45, 50, 55, 60, 65, 70, 75])

In [103]:
array_zero = np.zeros(5)
array_zero

array([0., 0., 0., 0., 0.])

In [104]:
array_one = np.ones(5)
array_one

array([1., 1., 1., 1., 1.])

In [109]:
array_random = np.random.rand(8)
array_random

array([0.37755654, 0.88966068, 0.95227268, 0.68334982, 0.72421771,
       0.05859703, 0.2333428 , 0.38062877])

In [113]:
array_integer = np.random.randint(10,101,size=8)
array_integer

array([29, 49, 56, 35, 96, 79, 67, 11])

*hint*

In [37]:
# Jawaban 1
import numpy as np

In [38]:
# Jawaban 2
my_array = np.array([100, 200, 300, 400, 500])
my_array

array([100, 200, 300, 400, 500])

In [50]:
# Jawaban 3
my_array_2 = np.arange(20, 80, 5 )
my_array_2

array([20, 25, 30, 35, 40, 45, 50, 55, 60, 65, 70, 75])

In [42]:
# Jawaban 4
array_zero = np.zeros(5)
array_zero

array([0., 0., 0., 0., 0.])

In [43]:
# Jawaban 5
array_one = np.ones(5)
array_one

array([1., 1., 1., 1., 1.])

In [44]:
# Jawaban 6
array_random = np.random.rand(8)
array_random

array([0.30778015, 0.01035122, 0.33132446, 0.05425464, 0.02387717,
       0.93840016, 0.02847023, 0.28684554])

In [110]:
# Jawaban 7
array_integer = np.random.randint(10, 100, size = 8)
array_integer

array([70, 27, 88, 59, 27, 45, 38, 27])

## Shape, Reshape, subsetting & Slicing

* 1. Check shape, size, dan dimensi dari my_array_2
* 2. Reshape my_array_2 menjadi 2D, dengan proporsi 3 rows dan 4 columns. simpan ke dalam variable dengan nama **my_2d_array**
* 3. Dari my_2d_array, pilih nilai 50 dan 70
* 4. Dari my_2d_array, pilih nilai 75

In [115]:
print(my_array_2.shape)
print(my_array_2.size)
print(my_array_2.ndim)

(12,)
12
1


In [117]:
my_2d_array = my_array_2.reshape((3,4))
my_2d_array

array([[20, 25, 30, 35],
       [40, 45, 50, 55],
       [60, 65, 70, 75]])

In [118]:
my_2d_array[1:3, 2]

array([50, 70])

In [119]:
my_2d_array[2,3]

75

*hint*

In [54]:
# Jawaban no. 1
print(my_array_2.shape)
print(my_array_2.ndim)
print(my_array_2.size)

(12,)
1
12


In [60]:
# Jawaban no. 2
my_2d_array = np.reshape(my_array_2, newshape = (3, 4))
my_2d_array

array([[20, 25, 30, 35],
       [40, 45, 50, 55],
       [60, 65, 70, 75]])

In [61]:
# Jawaban no. 4
my_2d_array[1:3, 2]

array([50, 70])

In [62]:
# Jawaban no. 4
my_2d_array[-1, -1]

75

## Array Operations

* 1. Tambahkan (*addition*) array_random dengan array_integer
* 2. Kalikan setiap element yang ada pada array_random dengan 100
* 3. dari my_2d_array cari nilai berikut: minimum, maximum, mean

In [120]:
array_random + array_integer

array([29.37755654, 49.88966068, 56.95227268, 35.68334982, 96.72421771,
       79.05859703, 67.2333428 , 11.38062877])

In [121]:
array_random * 100

array([37.75565391, 88.96606779, 95.22726777, 68.33498214, 72.42177142,
        5.85970283, 23.33427963, 38.06287739])

In [123]:
print(my_2d_array.min())
print(my_2d_array.max())
print(my_2d_array.mean())

20
75
47.5


In [64]:
# Jawaban no. 1

array_random + array_integer

array([63.30778015, 50.01035122, 98.33132446, 79.05425464, 31.02387717,
       24.93840016, 79.02847023, 23.28684554])

In [65]:
# Jawaban no. 2
array_random * 100

array([30.77801507,  1.03512187, 33.13244632,  5.4254641 ,  2.38771668,
       93.84001597,  2.84702328, 28.68455355])

In [71]:
# Jawaban no. 3
print(my_2d_array.min())
print(my_2d_array.max())
print(my_2d_array.mean())

20
75
47.5
